examples/dataset_example/src/dataset_example.cc

```diff
@@ -0,0 +1,239 @@
+// Copyright 2024 TGS
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//    http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file Provides an example of the basic usage of an MDIO Dataset.
+ * Goals:
+ * 1. Starting with an examlpe dataset, explore the variables and metadata.
+ * 2. Understand the in memory SharedArray and on disk representations of the
+ * data.
+ * 3. Read and write with a slicing operator.
+ * 4. Understand Future and Result operators.
+ * library.
+ */
+
+#include "dataset_example.h"
+
+#include <filesystem>
+
+#include "mdio/dataset.h"
+
+using Index = mdio::Index;
+using namespace mdio::dtypes;
+
+/**
+ * @brief This provides a basic MDIO example, covering its API usage and key
+ * concepts.
+ */
+absl::Status Run() {
+  /// Datasets create a number of zarr "files" which are directories
+  /// with json metadata.
+  std::filesystem::path temp_dir = std::filesystem::temp_directory_path();
+
+  std::string dataset_path = temp_dir / "zarrs/seismic_3d/";
+
+  auto json_spec = GetToyExample();
+
+  /// A dataset is used to represent a collection of data "on disk".
+  /// The MDIO json schema for this can be found here:
+  ///    mdio/dataset_schema.h
+  auto dataset_future = mdio::Dataset::from_json(json_spec, dataset_path,
+                                                 mdio::constants::kCreateClean);
```

Comment:
We should probably leave a comment here about the overwrite potential of using the `kCreateClean` option.

Reply:
Hi Brian, i've updated with changes.